src/agent/src/image.rs

```diff
@@ -57,9 +57,17 @@ pub struct ImageService {
 
 impl ImageService {
     pub fn new() -> Self {
+        let mut image_client = ImageClient::new(PathBuf::from(KATA_IMAGE_WORK_DIR));
+        let images = HashMap::new();
+        let registry_auth = &AGENT_CONFIG.image_registry_auth;
```

Comment:
Similar thought here, though maybe I'm misreading this.... but looks like you conditionally included this feature and so it may not be present here?

Reply:
Yep - you are right again, I need to work out how the scoping of `#[cfg(feature` can best to used in order to make this code less messy now!